src/Dapr.Client/DaprClient.cs

```diff
@@ -939,6 +940,293 @@ public async Task<StateEntry<TValue>> GetStateEntryAsync<TValue>(string storeNam
             string id,
             CancellationToken cancellationToken = default);
 
+        #region Cryptography
+
+        /// <summary>
+        /// Encrypts an array of bytes using the Dapr Cryptography encryption functionality.
+        /// </summary>
+        /// <param name="vaultResourceName">The name of the vault resource used by the operation.</param>
+        /// <param name="plainTextBytes">The bytes of the plaintext value to encrypt.</param>
+        /// <param name="algorithm">The name of the algorithm used to wrap the encryption key.</param>
+        /// <param name="keyName">The name of the key to use from the Vault for the encryption operation.</param>
+        /// <param name="dataEncryptionCipher">The name of the cipher to use for the encryption operation.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> that can be used to cancel the operation.</param>
+        /// <returns>An array of encrypted bytes.</returns>
+        [Obsolete("The API is currently not stable as it is in the Alpha stage. This attribute will be removed once it is stable.")]
+        public abstract Task<byte[]> EncryptAsync(string vaultResourceName, byte[] plainTextBytes,
```

Comment:
The number of arguments for these methods is rather high, which can make learning/use intimidating and error prone (i.e. inconsistencies from one call to the next).  I wonder if there'd be any benefit to moving some of the arguments to an "options" object to make the methods easier to digest as well as to promote the aggregation of related values (e.g. the algorithm, cipher, etc.) rather than being handed one-by-one to each encrypt call.

Reply:
This sort of fits into the next review note as well, but one of the goals stated for the reduced crypto interface footprint was that they wanted to simplify the approach as the "subtle" approach is too complex for the average person simply wanting to encrypt/decrypt something. What I was trying to do then was make it easy for anyone to pass in the barest number of arguments necessary to achieve that goal while also expose the few additional options available in the existing API for those mildly more advanced users.

The downside to the options approach is it simply shifts the complexity from the method arguments to options constructor overloads so that all the appropriately paired combinations are properly set. Yes, it'd limit the methods to two (one with options, one assuming some reasonable defaults), but it doesn't necessarily make using the options less intimidating.

That said, especially if there's interest in expanding the crypto building block to do more than the existing encrypt/decrypt (e.g. more of the "subtle" methods like wrapping keys or signing and verifying), I think separate options for each method would make a lot of sense to keep from ballooning the method space too broadly.

Reply:
I gave the options approach a whirl and I think you're right - it lends to a cleaner shape that can be augmented further later should more options emerge. Also added an option to configure the number of bytes to stream to the sidecar in each chunk.